clients/js/genjs.go

```diff
@@ -177,6 +177,21 @@ function responseLog(logger, req, res, err) {
   }
 }
 
+/**
+ * Takes a promise and uses the provided callback (if any) to handle promise
+ * resolutions and rejections
+ */
```

Comment:
Can you add the jsDoc `@private` tag so as not to expose these functions in the generated README? http://usejsdoc.org/tags-private.html